front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/PipelineControllerTck.groovy

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2014 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.front50.controllers
+
+import com.amazonaws.ClientConfiguration
+import com.amazonaws.services.s3.AmazonS3Client
+import com.netflix.spinnaker.front50.model.S3PipelineDAO
+import com.netflix.spinnaker.front50.model.pipeline.Pipeline
+import com.netflix.spinnaker.front50.model.pipeline.PipelineDAO
+import com.netflix.spinnaker.front50.pipeline.PipelineRepository
+import com.netflix.spinnaker.front50.utils.CassandraTestHelper
+import com.netflix.spinnaker.front50.utils.S3TestHelper
+import rx.schedulers.Schedulers
+import spock.lang.IgnoreIf
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Subject
+import spock.lang.Unroll
+
+import java.util.concurrent.Executors
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.setup.MockMvcBuilders
+
+abstract class PipelineControllerTck extends Specification {
```

Comment:
These tests used to be Mock-based which provided very little value.

I've started converting to be a lightweight TCK and test against `S3` and `Cassandra` DAOs.

Still some more tests that need flushing out but the confidence these will give us is significantly higher than anything else in Front50.